### PR TITLE
New Urdf Cheat Sheet Goodie

### PR DESCRIPTION
**Pull request by GuillermoUrcera**
## Testing & Review
To be completed by Community Leader (or DDG Staff) when reviewing Pull Request

**Pull Request**
- [ ] Title follows correct format (Specifies Instant Answer + Purpose)
- [ ] Description contains a valid Instant Answer Page Link (e.g. https://duck.co/ia/view/my_ia)

**Instant Answer Page** (for new Instant Answers)
- [ ] Instant Answer page is correctly filled out and contains:
    - The **Title** is appropriately named and formatted
    - The **IA topics** are present and appropriate
    - The **Description** is clear and coherent
    - **Source Name** exists if applicable
    - All **Example Queries** trigger on [Beta](https://beta.duckduckgo.com/)

**Code**
- [ ] Adheres to the [DuckDuckGo Style Guide](https://docs.duckduckhack.com/resources/code-style-guide.html)
- [ ] Behaviour is appropriately tested. If improvement, tests are adequately extended.
- [ ] There is no unnecessary files in place (such as editor config files)
- [ ] There is no API keys / secrets present
- [ ] Tests are passing (run `$ duckpan test <goodie_id>`)
    - Tester should report any failures

**Ready to merge?**

- [ ] Has this IA been deployed to and tested on [beta.duckduckgo.com](https://beta.duckduckgo.com/)?
- [ ] For larger commits, has this been approved be more than one community member?
- [ ] The number of reviews is appropriate for this type of PR
- [ ] The commit message is clear, coherent and fitting

**Pull Request Review Guidelines**: https://docs.duckduckhack.com/programming-mission/pr-review.html

---

Instant Answer Page: https://duck.co/ia/view/urdf_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: share/goodie/cheat_sheets/json/urdf.json [hunk 1]
```diff
@@ -0,0 +1,121 @@
+{
+    "id": "urdf_cheat_sheet",
+    "name": "URDF cheat sheet",
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/urdf.json]: Lets remove cheat sheet from name

File: share/goodie/cheat_sheets/json/urdf.json [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    "description": "Unified Robot Description Format (URDF), which is an XML format for representing a robot model.",
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/urdf.json]: remove period at end of description field.
Also we can remove the word 'which'

File: share/goodie/cheat_sheets/json/urdf.json [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    ],
+
+    "template_type": "code",
+
+    "section_order": [
+        "Joint types",
+        "Joint attributes",
+        "Joint elements",
+        "Link attributes",
+        "Link elements"
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/urdf.json]: Capitalize first letters of second word too

**Comment by sahildua2305** [comment]
@duckduckgo/duckduckhack-contributors need your input on this.

**Comment by adityatandon007** [comment]
**My Review**
@GuillermoUrcera This cheat sheet doesn't make sense. 
It only covers two aspect of urdf (xml format) 
namely the `joint` and `link`. The other thinks like `sensor`, `model`, `transmission`, `model state`, etc should be added to make it look like a cheat sheet.
One more thing you should change the `val` information to something meaningful instead of `required` or `optional`.

cc @sahildua2305 @pjhampton

**Comment by GuillermoUrcera** [comment]
@adityatandon007 
I made it so that it covered the most important aspects of an urdf file. This is the kind of thing I used to have to look up between different pages though the ROS documentation when I was using it.

The other elements you mentioned weren't added because:

- **sensor**: [the project was dropped ](http://wiki.ros.org/urdf/XML/sensor)
- **model**: [has no content, it redirects to **link** and **joint**](http://wiki.ros.org/urdf/XML/model)
- **model_state:** [work in progress, not actually used](http://wiki.ros.org/urdf/XML/model_state)
- **transmission:** I've not seen it used, I assumed it was not very important and that if it was someone later would add it.

I don't know what you consider a meaningful value for val since it's a hierarchical system and adding more detail would make the cheatsheet long and unwieldly. I considered it out of scope.

I made the cheatsheet as something someone who had to write a quick urdf file would find useful.

**Comment by mintsoft** [comment]
@adityatandon007 are you happy with the changes here?

**Comment by adityatandon007** [comment]
@mintsoft its fine for now and I know some of the projects are dropped or not implemented in applications but the cheat sheet just cover only **joint** element of the XML specification (http://wiki.ros.org/urdf/XML)
so this cheat sheet is basically a part of the whole XML Specification, that's the only point I want to make. We can have others opinions before we finally merge it.

**Comment by daxtheduck** [comment]
@GuillermoUrcera this is now merged! It should be live on DuckDuckGo within one week (or during the next release) and we'll be sure to notify you when it happens.

After your Instant Answer goes live for the World to use, you will be able to see [traffic information](http://docs.duckduckhack.com/maintaining/tools.html#monitor-your-instant-answer-page) about your IA as well as receive feedback directly from users. 
                
Please refer to the [maintenance guide and docs](http://docs.duckduckhack.com/maintaining/guidelines.html) to keep up with potential changes to your Instant Answer.
                
Thank you again for your help!

**Comment by mintsoft** [comment]
Thanks @GuillermoUrcera @sahildua2305 @adityatandon007

**Pull request merged**